tests/core/engine_adapter/integration/test_integration_duckdb.py

```diff
@@ -0,0 +1,72 @@
+import typing as t
+import pytest
+from threading import current_thread, Thread
+import random
+from sqlglot import exp
+
+from sqlmesh.core.config.connection import DuckDBConnectionConfig
+from sqlmesh.utils.connection_pool import ThreadLocalConnectionPool
+
+pytestmark = [pytest.mark.duckdb, pytest.mark.engine, pytest.mark.slow]
+
+
+@pytest.mark.parametrize("database", [None, "db.db"])
+def test_multithread_concurrency(tmp_path, database: t.Optional[str]):
```

Comment:
This is the test I wrote to explore DuckDB's concurrency. It's based on the example from their website.

It originally took _minutes_ to complete until I started sharing the same connection object across all threads, then it dropped down to a few seconds